### ISSUE TITLE
Add Chuvash voice

### ISSUE THREAD

**Issue by ftyers** [Olga-Yakovleva/RHVoice#67]
We have a project that collected 4 hours of sentence-aligned audio under a CC-BY-SA licence in Chuvash.[1] Is this enough to train RHVoice ? 

What are the steps we should go through? We trained Ossian and Merlin and have the system online.[2] But the more widely spread the data the better :)

1. https://github.com/ftyers/Turkic_TTS/tree/master/corpus/chv/speakers/chuvash_news
2. http://yukari.default.ftyers.uk0.bigv.io/~ossian/apertium-tts-web/

**Comment by Olga-Yakovleva** [comment]
Yes, it must be enough if voices for other statistical parametric speech 
synthesis systems have already been built from this data successfully.

The process of adding support for new languages and building new voices 
is not well organized in RHVoice. I didn't expect other people would 
want to do it and never got around to tidying things up and documenting 
the process. But the first step is adding support for the language 
itself: defining the phonemes and their properties, pronunciation rules, 
etc. I'm assuming your system is phoneme-based, not grapheme-based. 
Could you describe what format you are using for your language resources?

**Comment by ftyers** [comment]
@Olga-Yakovleva It's actually grapheme based, we found that for Chuvash a grapheme->phoneme model is not necessary, at least for training Ossian and Merlin.

The input format is a `.txt` file named by a unique ID with a list of sentences indexed in order, e.g. file [14459.txt](https://github.com/ftyers/Turkic_TTS/blob/master/corpus/chv/speakers/chuvash_news/transcripts/txt/14459.txt). The audio files are `.flac` and there is one audio file per `.txt` file. Each file has 5 seconds of silence at the beginning (for the noise profile) and 2 seconds of silence between each sentence, for example [14459.flac](https://github.com/ftyers/Turkic_TTS/blob/master/corpus/chv/speakers/chuvash_news/audio/orig/14459.flac).

The extraction, sentence alignment is described in our [README](https://github.com/ftyers/Turkic_TTS/blob/master/corpus/chv/speakers/chuvash_news/README.md). The output format of the data is basically `.flac` files with one sentence for the audio and then `.txt` files with one sentence for the text. 

I've uploaded a [tarball](http://ilazki.vm.bytemark.co.uk/~spectre/train-chuvash.tar.gz) of the final stuff in case you don't have time to run the extraction.

**Comment by Olga-Yakovleva** [comment]
Thank you for the files.

How well does your system handle Russian loanwords? I've built voices 
for other turkic languages spoken in the former Soviet Union. And it 
seems that Russian loanwords are a problem without a solution.

I assume I can map letters to phonemes directly. I know about turkic 
languages in general. But it might be useful to read about Chuvash 
phonology in particular. Could you recommend any sources of information 
except Wikipedia?

I'm interested in building the voice when I have the time. As it 
happens, I actually live in Chuvashia, though I don't speak the language 
at all. But I'd like to help other visually-impaired computer users here 
to read and write in their native language.

**Comment by ftyers** [comment]
You're welcome! (Тархасшӑн!) At least regarding Ossian/Merlin it works pretty well with Russian loan words. If you try something like "Манӑн мотоцикл пур, санӑн та мотоцикл пур-и ?" you will hear voicing on the та /da/ but not on мото /moto/, even though in Chuvash words intervocalic т should be pronounced something like /d/.

Regarding the mapping, it's largely straightforward, apart from consonant "voicing". There are no voiced consonants written in Chuvash, voicing is done in some phonological contexts but not marked in the orthography. It is also suprasegmental. So for example _эпӗ те_ "me too" is /ebɘ de/.

Oh, you live in Chuvashia! I should have mentioned, I was in Šupaškar last weekend and [gave a talk](https://vk.com/wall1644710_5370) about speech synthesis at the national library. :) [The slides](http://ilazki.thinkgeek.co.uk/~spectre/cmeld-tts.pdf) are online, and are basically the same ones I gave a the [CMLD workshop](http://lattice.cnrs.fr/cmld/program.html) in Paris last month.

I'm in Moscow, so am not infrequently in Chuvashia, if you'd like to meet up to discuss speech technology for Chuvash it would be great. You can find me on VK or by email.
